projects/jdwp/defs/schema.py

```diff
@@ -88,7 +88,7 @@ class Command:
     id: int
     out: Optional[Struct]
     reply: Optional[Struct]
-    error: Set[ErrorType]
+    error: Set[ErrorType] | None
```

Comment:
Set can be empty, so perhaps we don't need `None` here ?
1. It's good not to have two ways to represent the same fact: `None` and `set()` are two ways to represent that there are no errors associated with a command. It would be unfortunate if some commands used the first and some the second way.
2. it might lead to simpler code processing `Command`s if error cannot be None. For example, if we wanted to check whether an error code is valid then:

```
rc in command.error
```

is simpler than:

```
True if command.error is None else rc in command.error
```

At the same time, it's easy to check for there not being any error types:

```
if command.error:
    print("there are no errors allowed for this command")
```

Reply:
I have used an empty set: `set ()` for the command with no error.